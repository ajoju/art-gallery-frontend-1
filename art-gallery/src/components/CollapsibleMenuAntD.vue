<template>
    <div>
        <a-menu class="nav-menu-dropdown" mode="vertical">
            <a-sub-menu>
                <template #title>
                    <router-link to="/artworks" class="nav-link">
                        <span>Artworks</span>
                    </router-link>
                </template>
                <a-menu-item>
                    <router-link to="/artworkofday" class="nav-sub">
                        <span>Artwork of the Day</span>
                    </router-link>
                </a-menu-item>
                <a-menu-item><span class="nav-sub">Browse by Price</span></a-menu-item>
            </a-sub-menu>
            <a-menu-item>
                <div class="nav-tools-login" v-if="!account.user">
                    <router-link to="/login" class="nav-link">
                        <span>Log In</span>
                    </router-link>
                </div>
            </a-menu-item>
            <a-sub-menu>
                <template #title>
                    <router-link to="/culture" class="nav-link">
                        <span>Art & Culture</span>
                    </router-link>
                </template>
                    <a-menu-item><span class="nav-sub">Symbols</span></a-menu-item>
                    <a-menu-item>
                        <router-link to="/artistofday" class="nav-sub">
                            <span>Artist of the Day</span>
                        </router-link>
                    </a-menu-item>
                    <a-menu-item><span class="nav-sub">Aboriginal Art Facts</span></a-menu-item>
            </a-sub-menu>
            <a-menu-item>
                <div class="nav-tools-signup" v-if="!account.user">
                    <router-link to="/signup" class="nav-link">
                        <span>Sign Up</span>
                    </router-link>
                </div>
            </a-menu-item>
            <a-sub-menu>
                <template #title>
                    <router-link to="/exhibition" class="nav-link">
                        <span>Exhibitions</span>
                    </router-link>
                </template>
                <a-menu-item><span class="nav-sub">Current Exhibitions</span></a-menu-item>
                <a-menu-item><span class="nav-sub">Past Exhibitions</span></a-menu-item>
            </a-sub-menu>
        </a-menu>
    </div>
</template>


<script>
import { mapState } from 'vuex';

export default {
    name: "CollapsibleMenuAntD",
    computed: {
        ...mapState({
            account: state => state.account
        })
    }
}
</script>

<style scoped>
    .nav-sub:hover {
        color:var(--color--turquoise-light-hover);
        transition: .1s;
        transition-delay: 0;
        border-bottom: 2px solid;
    }
    .nav-menu-dropdown {
        display: grid;
        grid-template-columns: max-content auto;
    }
    .nav-sub {
        font-size: calc(12px + 2 * (100vw - 320px) / 1040);
    }
    .ant-menu::before {
        content: none;
    }
</style>