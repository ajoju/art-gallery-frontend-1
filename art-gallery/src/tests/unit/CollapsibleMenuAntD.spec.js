import { mount } from "@vue/test-utils";
import { expect, it, describe, beforeEach } from "vitest";
import CollapsibleMenuAntD from "@/components/CollapsibleMenuAntD.vue";

describe("CollapsibleMenuAntD", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(CollapsibleMenuAntD, {
      global: {
        mocks: {
          $store: {
            state: {
              account: {
                user: null,
              },
            },
          },
        },
      },
    });
  });

  it("renders correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders the menu with sub-menus", () => {
    const subMenus = wrapper.findAll("a-sub-menu");
    expect(subMenus.length).toBe(3);
  });

  it("renders the login and signup menu items", () => {
    const loginMenuItem = wrapper.find('a-menu-item[key="login"]');
    expect(loginMenuItem.exists()).toBe(true);

    const signupMenuItem = wrapper.find('a-menu-item[key="signup"]');
    expect(signupMenuItem.exists()).toBe(true);
  });

  it("renders the correct sub-menu titles", () => {
    const subMenuTitles = wrapper.findAll("template");
    expect(subMenuTitles[0].text()).toBe("artworks");
    expect(subMenuTitles[1].text()).toBe("art & culture");
    expect(subMenuTitles[2].text()).toBe("exhibitions");
  });

  it("renders the correct number of menu items in each sub-menu", () => {
    const subMenuItems = wrapper.findAll("a-menu-item");
    expect(subMenuItems.length).toBe(9);
  });
});
