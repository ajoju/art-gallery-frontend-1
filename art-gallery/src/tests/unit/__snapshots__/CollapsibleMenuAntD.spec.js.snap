// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CollapsibleMenuAntD > renders correctly 1`] = `
"<div data-v-b7bfe5a9=\\"\\">
  <a-menu data-v-b7bfe5a9=\\"\\" openkeys=\\"\\" selectedkeys=\\"\\" class=\\"nav-menu-dropdown\\" mode=\\"inline\\">
    <div data-v-b7bfe5a9=\\"\\">
      <a-sub-menu data-v-b7bfe5a9=\\"\\">
        <a-menu-item data-v-b7bfe5a9=\\"\\">
          <router-link data-v-b7bfe5a9=\\"\\" to=\\"/artworkofday\\" class=\\"nav-sub\\"><span data-v-b7bfe5a9=\\"\\">Artwork of the Day</span></router-link>
        </a-menu-item>
        <a-menu-item data-v-b7bfe5a9=\\"\\">
          <router-link data-v-b7bfe5a9=\\"\\" to=\\"/artworks\\" class=\\"nav-sub\\"><span data-v-b7bfe5a9=\\"\\">List of Artworks</span></router-link>
        </a-menu-item>
      </a-sub-menu>
      <a-sub-menu data-v-b7bfe5a9=\\"\\">
        <a-menu-item data-v-b7bfe5a9=\\"\\"><span data-v-b7bfe5a9=\\"\\" class=\\"nav-sub\\">Symbols</span></a-menu-item>
        <a-menu-item data-v-b7bfe5a9=\\"\\">
          <router-link data-v-b7bfe5a9=\\"\\" to=\\"/artistofday\\" class=\\"nav-sub\\"><span data-v-b7bfe5a9=\\"\\">Artist of the Day</span></router-link>
        </a-menu-item>
        <a-menu-item data-v-b7bfe5a9=\\"\\"><span data-v-b7bfe5a9=\\"\\" class=\\"nav-sub\\">Aboriginal Art Facts</span></a-menu-item>
      </a-sub-menu>
      <a-sub-menu data-v-b7bfe5a9=\\"\\">
        <a-menu-item data-v-b7bfe5a9=\\"\\">
          <router-link data-v-b7bfe5a9=\\"\\" to=\\"/exhibition\\" class=\\"nav-sub\\"><span data-v-b7bfe5a9=\\"\\">Current Exhibitions</span></router-link>
        </a-menu-item>
        <a-menu-item data-v-b7bfe5a9=\\"\\"><span data-v-b7bfe5a9=\\"\\" class=\\"nav-sub\\">Past Exhibitions</span></a-menu-item>
      </a-sub-menu>
    </div>
    <div data-v-b7bfe5a9=\\"\\">
      <a-menu-item data-v-b7bfe5a9=\\"\\">
        <div data-v-b7bfe5a9=\\"\\" class=\\"nav-tools-login\\">
          <router-link data-v-b7bfe5a9=\\"\\" to=\\"/login\\" class=\\"nav-link\\"><span data-v-b7bfe5a9=\\"\\">Log In</span></router-link>
        </div>
      </a-menu-item>
      <a-menu-item data-v-b7bfe5a9=\\"\\">
        <div data-v-b7bfe5a9=\\"\\" class=\\"nav-tools-signup\\">
          <router-link data-v-b7bfe5a9=\\"\\" to=\\"/signup\\" class=\\"nav-link\\"><span data-v-b7bfe5a9=\\"\\">Sign Up</span></router-link>
        </div>
      </a-menu-item>
    </div>
  </a-menu>
</div>"
`;
